src/afpacket_plugin/afpacket_collector.cpp

```diff
@@ -259,8 +264,12 @@ int setup_socket(std::string interface_name, int fanout_group_id) {
         struct block_desc *pbd = (struct block_desc *) rd[current_block_num].iov_base;
  
         if ((pbd->h1.block_status & TP_STATUS_USER) == 0) {
-            poll(&pfd, 1, -1);
-
+            try {
+                poll(&pfd, 1, 500);
+                boost::this_thread::interruption_point();
```

Comment:
You're making changes in extremely critical to delays path. interruption_point has non-zero overhead. 

I skipped this code when filled all parts with interruption_point for a reason.

But if you could confirm that this code still could do 9-10 millions of packets per second: kukuruku.co/post/capturing-packets-in-linux-at-a-speed-of-millions-of-packets-per-second-without-using-third-party-libraries/ we could move forward with it :)